src/linkcreator.cpp

```diff
@@ -60,17 +60,19 @@ Data::TypeReference LinkCreator::getSymbolTypeReference(const QTextCursor &curso
     if (!m_textDocument.characterAt(cursor.position()).isLetterOrNumber())
         return Data::TypeReference();
 
-    const auto file = ParsedDataStorage::instance()->getAnyFileForPath(m_documentPath);
+    const auto file = m_storage->getAnyFileForPath(m_documentPath);
 
-    return getTypeReference(file, cursor.blockNumber() + 1, cursor.columnNumber());
+    return file != nullptr
+            ? getTypeReference(file, cursor.blockNumber() + 1, cursor.columnNumber())
+            : Data::TypeReference();
```

Comment:
I'd move this check into `getTypeReference`, so if anyone will call it in future it will already be safe